core/src/test/java/com/datastrato/gravitino/storage/relational/TestRelationalEntityStore.java

```diff
@@ -172,22 +211,101 @@ public void testPutAndList() throws IOException {
     assertEquals(2, metalakes.size());
     assertTrue(checkMetalakeEquals(metalake1, metalakes.get(0)));
     assertTrue(checkMetalakeEquals(metalake2, metalakes.get(1)));
+
+    // catalog
+    CatalogEntity catalog1 =
+        createCatalog(
+            1L, "test_catalog1", Namespace.ofCatalog(metalake1.name()), "this is catalog 1");
+    CatalogEntity catalog2 =
+        createCatalog(
+            2L, "test_catalog2", Namespace.ofCatalog(metalake1.name()), "this is catalog 2");
+    List<CatalogEntity> beforeCatalogList =
+        entityStore.list(catalog1.namespace(), CatalogEntity.class, Entity.EntityType.CATALOG);
+    assertNotNull(beforeCatalogList);
+    assertEquals(0, beforeCatalogList.size());
+
+    entityStore.put(catalog1, false);
+    entityStore.put(catalog2, false);
+    List<CatalogEntity> catalogEntities =
+        entityStore.list(catalog1.namespace(), CatalogEntity.class, Entity.EntityType.CATALOG);
+    assertNotNull(catalogEntities);
+    assertEquals(2, catalogEntities.size());
+    assertTrue(checkCatalogEquals(catalog1, catalogEntities.get(0)));
```

Comment:
You may need to sort the results as it can't guarantee the order.

Reply:
Done